bankAccount-system/index.js

```diff
@@ -0,0 +1,150 @@
+        this._lastName = lastName;
+        this._accountHolder = `${firstName} ${lastName}`;
+        this._balance = balance;
+        this._transactions = [];
+        this._dailyWithdrawalLimit = dailyWithdrawalLimit
+        this._withdrawnToday = 0;
+        this._lastWithdrawalDate = new Date().getDate();
+    }
+
+    get accountNumber() {
+        return this._accountNumber;
+    }
+
+    get firstName () {
+        return this._firstName;
+    }
+
+    get lastName () {
+        return this._lastName;
+    }
+
+    get accountHolder () {
+        return this._accountHolder;
+    }
+
+    get balance () {
+        return this._balance;
+    }
+
+    deposit(amount) {
+        if (amount <= 10) {
+            throw new Error('Invalid amount entered. Minimum depost is 10');
+        }
+        else {
+            this._balance += amount;
+            this._transactions.push(new Transaction('Deposit', amount));
+            return `${this._accountHolder}, ${amount} has been deposited to your account. Your new balance is ${this._balance}`
+        }
+    }
+    
+    withdraw(amount) {
+        const today = new Date().getDate();
+
+        // Checks if its a new day and reset amount withdrawn today if necessary
+        if (today !== this._lastWithdrawalDate){
+            this._withdrawnToday = 0;
+            this._lastWithdrawalDate = today;
+        }
+
+        // Checks if there is enough balance for withdrawal
+        if (amount > this._balance) {
+            throw new Error('Insufficient funds');
+        }
+
+        // Checks if the amount exceeds the daily withdrawal limit
+        if ((this._withdrawnToday + amount) > this._dailyWithdrawalLimit) {
+            throw new Error('Exceeds Daily Withdrawal Limit')
+        }
+
+        this._balance -= amount;
+        this._withdrawnToday += amount;
+        this._transactions.push(new Transaction('Withdrawal', amount));
+        return `${this._accountHolder}, ${amount} has been withdrawn from your account. Your new balance is ${this._balance}`
+       
+    }
+
+    transfer(amount, recipientAccount) {
+
+        // Checks if the amount is positive
+        if (amount < 0) {
+            throw new Error('Invalid transfer amount');
+        }
+        
+        // Checks if there is enough balance for transfer
+        if (amount > this._balance) {
+            throw new Error('Insufficient funds');
+        }
+
+        // Checks if the amount exceeds the daily withdrawal limit
+        if ((this._withdrawnToday + amount) > this._dailyWithdrawalLimit) {
+            throw new Error('Exceeds Daily Withdrawal Limit')
+        }
+
+        this._balance -= amount;
+        this._withdrawnToday += amount;
+        recipientAccount.deposit(amount);
+        this._transactions.push(new Transaction('Transfer', amount));
+        return `You transferred ${amount} from this ${this._accountNumber} to ${recipientAccount.accountNumber}.`;
+    }
+
+    getTransactions() {
```

Comment:
as partaining to the third assignment, this feature should return an array of javascript object. NOTE javascript objects not javascript classes